lib/ts/types.ts

```diff
@@ -528,8 +528,13 @@ export type SuperTokensPlugin = {
     version?: string;
     compatibleAuthReactSDKVersions?: string | string[]; // match the syntax of the engines field in package.json
     compatibleWebJSSDKVersions?: string | string[]; // match the syntax of the engines field in package.json
+    init?: (
+        config: Omit<SuperTokensConfig, "experimental" | "recipeList">,
+        allPlugins: Pick<SuperTokensPlugin, "id" | "version" | "exports">[],
+        sdkVersion: string
+    ) => Promise<void> | void;
```

Comment:
I don't think there's a usecase for an async init function, so we should just drop the promise part for now.